runtime/doc/various.txt

```diff
@@ -585,6 +585,28 @@ to look up the value of 'commentstring' corresponding to the cursor position.
 (This can be different from the buffer's 'commentstring' in case of
 |treesitter-language-injections|.)
 
+The comment module can take a custom hook to allow for more control when
+determining the 'commentstring'. For example, the following can be used to get
+more accurate comments in JSX-family files (assuming the appropriate
+|treesitter| parser has been installed): >lua
+
+    require('vim._comment').cs_hook = function(row, col)
```

Comment:
"hook" is not a name used anywhere else in vim/nvim. Use existing names/conventions, to help discoverability.

This would typically be modeled as an "event" (autocmd) or a callback (`on_foo`).